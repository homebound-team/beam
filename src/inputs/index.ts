export * from "./Checkbox";
export * from "./CheckboxGroup";
export * from "./ChipSelectField";
export * from "./DateFields";
export * from "./MultiLineSelectField";
export * from "./MultiSelectField";
export * from "./NumberField";
export type { NumberFieldProps } from "./NumberField";
export { RadioGroupField } from "./RadioGroupField";
export type { RadioFieldOption, RadioGroupFieldProps } from "./RadioGroupField";
export * from "./RichTextField";
export * from "./SelectField";
export * from "./Switch";
export * from "./TextAreaField";
export * from "./TextField";
export type { TextFieldApi } from "./TextField";
export * from "./ToggleButton";
export * from "./ToggleChipGroup";
export type { Value } from "./Value";
