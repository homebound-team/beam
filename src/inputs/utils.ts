export function getFieldWidth(fullWidth: boolean | undefined) {
  return fullWidth ? "100%" : "550px";
}
