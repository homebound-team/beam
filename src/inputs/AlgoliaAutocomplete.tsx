import { Key, ReactNode, useRef } from "react";
import { useComboBox, useOverlayPosition } from "react-aria";
import { Item, useComboBoxState } from "react-stately";
import { Icon } from "src/components";
import { Popover } from "src/components/internal";
import { PresentationFieldProps } from "src/components/PresentationContext";
import { ListBox } from "src/inputs/internal/ListBox";
import { TextFieldBase, TextFieldBaseProps } from "src/inputs/TextFieldBase";
import { valueToKey } from "src/inputs/Value";

interface AutocompleteProps<T>
  extends Pick<PresentationFieldProps, "labelStyle">,
    Pick<TextFieldBaseProps<any>, "label" | "clearable" | "startAdornment"> {
  onSelect: (item: T) => void;
  getOptionMenuLabel?: (o: T) => ReactNode;
  getOptionLabel: (o: T) => string;
  getOptionValue: (o: T) => string;
  onChange: (value: string) => void;
  value: string;
  options: T[];
  placeholder?: string;
  disabled?: boolean;
}

export function AlgoliaAutocomplete<T extends object>(props: AutocompleteProps<T>) {
  const { onSelect, getOptionLabel, getOptionValue, getOptionMenuLabel, onChange, value, options, ...others } = props;

  const comboBoxProps = {
    allowsEmptyCollection: true,
    onInputChange: onChange,
    children: (item: T) => (
      <Item key={getOptionValue(item)} textValue={getOptionValue(item)}>
        {getOptionMenuLabel ? getOptionMenuLabel(item) : getOptionLabel(item)}
      </Item>
    ),
    onSelectionChange: (key: Key) => {
      const selectedItem = options.find((i) => getOptionValue(i) === key);
      if (selectedItem) {
        onChange(getOptionLabel(selectedItem));
        onSelect(selectedItem);
      }
    },
    inputValue: value,
    items: options,
    ...others,
  };

  const state = useComboBoxState<T>(comboBoxProps);
  const inputWrapRef = useRef<HTMLDivElement>(null);
  const inputRef = useRef(null);
  const listBoxRef = useRef(null);
  const popoverRef = useRef(null);
  const { inputProps, listBoxProps, labelProps } = useComboBox(
    { ...comboBoxProps, inputRef, listBoxRef, popoverRef },
    state,
  );

  // useOverlayPosition moves the overlay to the top of the DOM to avoid any z-index issues. Uses the `targetRef` to DOM placement
  const { overlayProps: positionProps } = useOverlayPosition({
    targetRef: inputWrapRef,
    overlayRef: popoverRef,
    scrollRef: listBoxRef,
    isOpen: state.isOpen,
    onClose: state.close,
    placement: "bottom left",
  });

  positionProps.style = {
    ...positionProps.style,
    width: inputWrapRef?.current?.clientWidth,
    // Ensures the menu never gets too small.
    minWidth: 200,
  };

  return (
    <div>
      <TextFieldBase
        inputRef={inputRef}
        inputWrapRef={inputWrapRef}
        inputProps={inputProps}
        labelProps={labelProps}
        onChange={(v) => onChange(v ?? "")}
        clearable
        startAdornment={<Icon icon="search" />}
        {...others}
      />
      {state.isOpen && options.length > 0 && (
        <Popover
          triggerRef={inputRef}
          popoverRef={popoverRef}
          positionProps={positionProps}
          onClose={() => state.close()}
          isOpen={state.isOpen}
          minWidth={200}
        >
          <ListBox
            {...listBoxProps}
            positionProps={positionProps}
            state={state}
            listBoxRef={listBoxRef}
            getOptionValue={(o) => valueToKey(getOptionValue(o))}
            getOptionLabel={getOptionLabel}
          />
        </Popover>
      )}
    </div>
  );
}
