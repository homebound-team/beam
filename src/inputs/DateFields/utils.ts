import { format as dateFnsFormat, parse as dateFnsParse, isDate } from "date-fns";
import { DateRange } from "src/types";

export type DateFieldModeTuple = readonly ["range", DateRange] | readonly ["single", Date];
export type DateFieldMode = "single" | "range";

export const dateFormats = {
  short: "MM/dd/yy",
  medium: "EEE, MMM d",
  long: "EEEE LLLL d, uuuu",
};

export function getDateFormat(format: keyof typeof dateFormats | undefined) {
  return format ? dateFormats[format] : dateFormats.short;
}

export function formatDate(date: Date | undefined, format: string) {
  if (!date) return "";
  return dateFnsFormat(date as Date, format);
}

export function formatDateRange(date: DateRange | undefined, format: string) {
  if (!date) return "";
  const { from, to } = date as DateRange;
  const fromFormatted = from ? dateFnsFormat(from, format) : "";
  const toFormatted = to ? dateFnsFormat(to, format) : "";
  // return `undefined` if both dates are improperly formatted
  return !fromFormatted && !toFormatted ? undefined : `${fromFormatted} - ${toFormatted}`;
}

export function parseDate(str: string, format: string): Date | undefined {
  return parseDateString(str, format);
}

export function parseDateRange(str: string, format: string): DateRange | undefined {
  const [from = "", to = ""] = str.split("-");
  const fromDate = parseDateString(from.trim(), format);
  const toDate = parseDateString(to.trim(), format);
  // In the event the user mixes up the to/from dates then correct them.
  if (toDate && fromDate && toDate < fromDate) {
    return { from: toDate, to: fromDate };
  }
  // If both dates are undefined, return undefined rather than { from: undefined; to: undefined }
  if (toDate === undefined && fromDate === undefined) {
    return undefined;
  }
  return { from: fromDate, to: toDate };
}

function parseDateString(str: string, format: string): Date | undefined {
  // Copy/pasted from react-day-picker so that typing "2/2/2" doesn't turn into "02/02/0002"
  const split = str.split("/");
  if (split.length !== 3) {
    return undefined;
  }
  // Wait for the year to be 2 chars
  if (split[2].length !== 2) {
    return undefined;
  }
  const month = parseInt(split[0], 10) - 1;
  const day = parseInt(split[1], 10);
  const year = parseInt(split[2], 10);
  // This is also ~verbatim copy/pasted from react-day-picker
  if (
    isNaN(year) ||
    String(year).length > 4 ||
    isNaN(month) ||
    isNaN(day) ||
    day <= 0 ||
    day > 31 ||
    month < 0 ||
    month >= 12
  ) {
    return undefined;
  }

  const parsed = dateFnsParse(str, format, new Date());
  if (!isValidDate(parsed)) {
    return undefined;
  }
  return parsed;
}

export function isValidDate(d: Date | undefined): boolean {
  return d !== undefined && isDate(d) && d.toString() !== "Invalid Date";
}
