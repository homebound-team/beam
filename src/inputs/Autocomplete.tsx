import { Key, ReactNode, useRef } from "react";
import { useComboBox, useOverlayPosition } from "react-aria";
import { Item, useAsyncList, useComboBoxState } from "react-stately";
import { Popover } from "src/components/internal";
import { ListBox } from "src/inputs/internal/ListBox";
import { TextFieldBase } from "src/inputs/TextFieldBase";
import { valueToKey } from "src/inputs/Value";

interface AutocompleteProps<T> {
  onSelect: (item: T) => void;
  onSearch: (searchStr: string | undefined) => Promise<T[]>;
  getOptionMenuLabel?: (o: T) => ReactNode;
  getOptionLabel: (o: T) => string;
  getOptionValue: (o: T) => string;
  label: string;
}

export function Autocomplete<T extends object>(props: AutocompleteProps<T>) {
  const { onSearch, onSelect, getOptionLabel, getOptionValue, getOptionMenuLabel, label } = props;
  const list = useAsyncList<T>({
    async load({ filterText }) {
      const items = filterText ? await onSearch(filterText) : [];
      return { items };
    },
  });

  const comboBoxProps = {
    allowsEmptyCollection: true,
    onInputChange: list.setFilterText,
    children: (item: T) => (
      <Item key={getOptionValue(item)} textValue={getOptionValue(item)}>
        {getOptionMenuLabel ? getOptionMenuLabel(item) : getOptionLabel(item)}
      </Item>
    ),
    onSelectionChange: (key: Key) => {
      const selectedItem = list.items.find((i) => getOptionValue(i) === key);
      if (selectedItem) {
        list.setFilterText(getOptionLabel(selectedItem));
        onSelect(selectedItem);
      }
    },
    inputValue: list.filterText,
    items: list.items,
    label,
  };

  const state = useComboBoxState<T>(comboBoxProps);
  const inputWrapRef = useRef<HTMLDivElement>(null);
  const inputRef = useRef(null);
  const listBoxRef = useRef(null);
  const popoverRef = useRef(null);
  const { inputProps, listBoxProps, labelProps } = useComboBox(
    { ...comboBoxProps, inputRef, listBoxRef, popoverRef },
    state,
  );

  // useOverlayPosition moves the overlay to the top of the DOM to avoid any z-index issues. Uses the `targetRef` to DOM placement
  const { overlayProps: positionProps } = useOverlayPosition({
    targetRef: inputWrapRef,
    overlayRef: popoverRef,
    scrollRef: listBoxRef,
    isOpen: state.isOpen,
    onClose: state.close,
    placement: "bottom left",
  });

  positionProps.style = {
    ...positionProps.style,
    width: inputWrapRef?.current?.clientWidth,
    // Ensures the menu never gets too small.
    minWidth: 200,
  };

  return (
    <div>
      <TextFieldBase
        inputRef={inputRef}
        inputWrapRef={inputWrapRef}
        inputProps={inputProps}
        labelProps={labelProps}
        label={label}
        onChange={(v) => list.setFilterText(v ?? "")}
        clearable
      />
      {state.isOpen && list.items.length > 0 && (
        <Popover
          triggerRef={inputRef}
          popoverRef={popoverRef}
          positionProps={positionProps}
          onClose={() => state.close()}
          isOpen={state.isOpen}
          minWidth={200}
        >
          <ListBox
            {...listBoxProps}
            positionProps={positionProps}
            state={state}
            listBoxRef={listBoxRef}
            getOptionValue={(o) => valueToKey(getOptionValue(o))}
            getOptionLabel={getOptionLabel}
            loading={list.isLoading}
          />
        </Popover>
      )}
    </div>
  );
}
