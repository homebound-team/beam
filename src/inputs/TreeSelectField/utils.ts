import { Node } from "@react-types/shared";
import { Key } from "react";
import { Value } from "src/inputs/Value";

type FoundOption<O> = { option: NestedOption<O>; parents: NestedOption<O>[] };
export type NestedOption<O> = O & { children?: NestedOption<O>[] };
export type NestedOptionsOrLoad<O> =
  | NestedOption<O>[]
  | { current: NestedOption<O>[]; load: () => Promise<{ options: NestedOption<O>[] }> };
export type LeveledOption<O> = [NestedOption<O>, number];

export type TreeFieldState<O> = {
  inputValue: string;
  filteredOptions: LeveledOption<O>[];
  selectedKeys: Key[];
  selectedOptions: NestedOption<O>[];
  allOptions: NestedOption<O>[];
  optionsLoading: boolean;
  allowCollapsing: boolean;
};

export type TreeSelectResponse<O, V extends Value> = {
  all: { values: V[]; options: O[] };
  leaf: { values: V[]; options: O[] };
  root: { values: V[]; options: O[] };
};

/** Finds an option by Key, and returns it + any parents. */
export function findOption<O, V extends Value>(
  options: NestedOption<O>[],
  key: Key,
  getOptionValue: (o: O) => V,
): FoundOption<O> | undefined {
  // This is technically an array of "maybe FoundRow"
  const todo: FoundOption<O>[] = options.map((option) => ({ option, parents: [] }));
  while (todo.length > 0) {
    const curr = todo.pop()!;
    if (getOptionValue(curr.option) === key) {
      return curr;
    } else if (curr.option.children) {
      // Search our children and pass along us as the parent
      todo.push(...curr.option.children.map((option) => ({ option, parents: [...curr.parents, curr.option] })));
    }
  }
  return undefined;
}

export function flattenOptions<O>(o: NestedOption<O>): NestedOption<O>[] {
  return [o, ...(o.children?.length ? o.children.flatMap((oc: NestedOption<O>) => flattenOptions(oc)) : [])];
}

export function isLeveledOption<O>(option: LeveledOption<O> | any): option is LeveledOption<O> {
  return Array.isArray(option) && option.length === 2 && typeof option[1] === "number";
}

export function isLeveledNode<O>(node: Node<O> | Node<LeveledOption<O>>): node is Node<LeveledOption<O>> {
  return isLeveledOption(node.value);
}
