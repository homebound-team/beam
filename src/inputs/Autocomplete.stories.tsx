import { action } from "@storybook/addon-actions";
import { Meta, StoryFn } from "@storybook/react";
import { within } from "@storybook/testing-library";
import { useState } from "react";
import { useFilter } from "react-aria";
import { Css } from "src/Css";
import { Autocomplete } from "src/inputs/Autocomplete";

export default {
  component: Autocomplete,
} as Meta;

export const Example: StoryFn = Template.bind({});
Example.parameters = { chromatic: { disableSnapshot: true } };

export const MenuOpen: StoryFn = Template.bind({});
MenuOpen.play = async ({ canvasElement }: { canvasElement: HTMLElement }) => {
  const canvas = within(canvasElement);
  canvas.getByTestId("heroes").focus();
};

function Template() {
  const { contains } = useFilter({ sensitivity: "base" });
  const [value, setValue] = useState<string>();
  const allOptions = [
    { label: "Iron man", imgSrc: "tony-stark.jpg" },
    { label: "Captain Marvel", imgSrc: "captain-marvel.jpg" },
    { label: "Captain America", imgSrc: "captain-america.jpg" },
    { label: "Thor", imgSrc: "thor.jpg" },
    { label: "Black Widow", imgSrc: "/black-widow.jpg" },
  ];
  const [options, setOptions] = useState(allOptions);

  return (
    <Autocomplete<TestOption>
      label="Heroes"
      labelStyle="hidden"
      getOptionValue={(o) => o.label}
      getOptionLabel={(o) => o.label}
      getOptionMenuLabel={(o) => (
        <div css={Css.df.aic.gap1.$}>
          <span css={Css.wPx(24).hPx(24).br100.overflowHidden.$}>
            <img src={o.imgSrc} alt={o.label} css={Css.w100.h100.objectCover.$} />
          </span>
          {o.label}
        </div>
      )}
      onSelect={action("onSelect")}
      options={options}
      onInputChange={(value) => {
        setValue(value);
        setOptions(allOptions.filter((o) => contains(o.label, value ?? "")));
      }}
      value={value}
    />
  );
}

type TestOption = {
  label: string;
  imgSrc: string;
};
