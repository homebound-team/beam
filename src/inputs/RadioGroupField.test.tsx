import { RadioGroupField } from "src/inputs";
import { click, render } from "src/utils/rtl";

describe("RadioGroupField", () => {
  it("has data-testids for its options", async () => {
    const r = await render(
      <RadioGroupField
        label={"Favorite cheese"}
        value="a"
        onChange={() => {}}
        options={[
          { value: "a", label: "Asiago" },
          { value: "b", label: "Burratta" },
        ]}
      />,
    );
    click(r.favoriteCheese_a);
  });

  it("should disable only first option", async () => {
    const r = await render(
      <RadioGroupField
        label={"Favorite cheese"}
        value="a"
        onChange={() => {}}
        options={[
          { value: "a", label: "Asiago", disabled: true },
          { value: "b", label: "Burratta" },
        ]}
      />,
    );
    const radioInput = r.container.querySelector(`[data-testid="favoriteCheese_a"]`)!;
    expect(radioInput).toBeDisabled();
  });

  it("should disable first option and have a tooltip", async () => {
    const r = await render(
      <RadioGroupField
        label={"Favorite cheese"}
        value="a"
        onChange={() => {}}
        options={[
          { value: "a", label: "Asiago", disabled: "some reason" },
          { value: "b", label: "Burratta" },
        ]}
      />,
    );
    const tooltip = r.container.querySelector(`[data-testid="tooltip"]`)!;
    expect(tooltip).toBeInTheDocument();
    expect(tooltip).toHaveAttribute("title", "some reason");
  });

  it("should have a group name", async () => {
    const r = await render(
      <>
        <RadioGroupField
          name="favorite-cheese-group"
          label="cheese option 1"
          value="a"
          onChange={() => {}}
          options={[{ value: "a", label: "Asiago" }]}
        />
        <RadioGroupField
          name="favorite-cheese-group"
          label="cheese option 2"
          value="b"
          onChange={() => {}}
          options={[{ value: "b", label: "Burratta" }]}
        />
      </>,
    );
    expect(r.getByTestId("cheeseOption_1_a")).toHaveAttribute("name", "favorite-cheese-group");
    expect(r.getByTestId("cheeseOption_2_b")).toHaveAttribute("name", "favorite-cheese-group");
  });
});
