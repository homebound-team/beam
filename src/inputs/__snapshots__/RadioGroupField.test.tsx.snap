// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroupField renders with just labels 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 0;
  min-width: 0;
}

.emotion-1 {
  padding-left: 0px;
  padding-right: 0px;
}

.emotion-2 {
  max-width: 320px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.emotion-3 {
  color: rgba(100,100,100,1);
  display: inline-block;
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  margin-bottom: 4px;
}

.emotion-4 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
}

.emotion-5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 100%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  width: 16px;
  outline: 0px solid transparent;
  background-color: currentColor;
  border-color: currentColor;
  border-style: solid;
  border-width: 1px;
  color: rgba(3,105,161,1);
  -webkit-transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms,margin 200ms;
  transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms,margin 200ms;
  -webkit-background-position: center;
  background-position: center;
  background-repeat: no-repeat;
  -webkit-background-size: 100% 100%;
  background-size: 100% 100%;
  background-image: url("data:image/svg+xml,%3csvg viewBox='0 0 16 16' fill='white' xmlns='http://www.w3.org/2000/svg'%3e%3ccircle cx='8' cy='8' r='4'/%3e%3c/svg%3e");
  margin-right: 8px;
  margin-top: 2px;
}

.emotion-6 {
  color: rgba(53,53,53,1);
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
}

.emotion-8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 100%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  width: 16px;
  outline: 0px solid transparent;
  background-color: rgba(254,254,254,1);
  border-color: rgba(221,220,220,1);
  border-style: solid;
  border-width: 1px;
  color: rgba(3,105,161,1);
  -webkit-transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms,margin 200ms;
  transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms,margin 200ms;
  cursor: pointer;
  margin-right: 8px;
  margin-top: 2px;
}

.emotion-10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: auto;
  padding-left: 0px;
  padding-right: 0px;
}

<body>
  <div>
    <div
      data-overlay-container="true"
    >
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            <label
              class="emotion-3"
              data-testid="favoriteCheese_label"
              id="react-aria-2"
            >
              Favorite cheese
            </label>
            <div
              aria-labelledby="react-aria-2"
              aria-orientation="vertical"
              id="react-aria-1"
              role="radiogroup"
            >
              <label
                class="emotion-4"
              >
                <input
                  aria-labelledby="radio-group-1-a-label"
                  checked=""
                  class="emotion-5"
                  data-testid="favoriteCheese_a"
                  name="react-aria-3"
                  tabindex="0"
                  type="radio"
                  value="a"
                />
                <div>
                  <div
                    class="emotion-6"
                    id="radio-group-1-a-label"
                  >
                    Asiago
                  </div>
                </div>
              </label>
              <label
                class="emotion-4"
              >
                <input
                  aria-labelledby="radio-group-1-b-label"
                  class="emotion-8"
                  data-testid="favoriteCheese_b"
                  name="react-aria-3"
                  tabindex="0"
                  type="radio"
                  value="b"
                />
                <div>
                  <div
                    class="emotion-6"
                    id="radio-group-1-b-label"
                  >
                    Burratta
                  </div>
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          class="emotion-10"
        >
          <div
            style="height: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`RadioGroupField renders with just labels and description 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 0;
  min-width: 0;
}

.emotion-1 {
  padding-left: 0px;
  padding-right: 0px;
}

.emotion-2 {
  max-width: 344px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.emotion-3 {
  color: rgba(100,100,100,1);
  display: inline-block;
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  margin-bottom: 4px;
}

.emotion-4 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 8px;
}

.emotion-5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 100%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  width: 16px;
  outline: 0px solid transparent;
  background-color: currentColor;
  border-color: currentColor;
  border-style: solid;
  border-width: 1px;
  color: rgba(3,105,161,1);
  -webkit-transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms,margin 200ms;
  transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms,margin 200ms;
  -webkit-background-position: center;
  background-position: center;
  background-repeat: no-repeat;
  -webkit-background-size: 100% 100%;
  background-size: 100% 100%;
  background-image: url("data:image/svg+xml,%3csvg viewBox='0 0 16 16' fill='white' xmlns='http://www.w3.org/2000/svg'%3e%3ccircle cx='8' cy='8' r='4'/%3e%3c/svg%3e");
  margin-right: 8px;
  margin-top: 2px;
}

.emotion-6 {
  color: rgba(53,53,53,1);
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
}

.emotion-8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 100%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  width: 16px;
  outline: 0px solid transparent;
  background-color: rgba(254,254,254,1);
  border-color: rgba(221,220,220,1);
  border-style: solid;
  border-width: 1px;
  color: rgba(3,105,161,1);
  -webkit-transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms,margin 200ms;
  transition: background-color 200ms,border-color 200ms,box-shadow 200ms,left 200ms,right 200ms,margin 200ms;
  cursor: pointer;
  margin-right: 8px;
  margin-top: 2px;
}

.emotion-10 {
  color: rgba(100,100,100,1);
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
}

.emotion-11 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: auto;
  padding-left: 0px;
  padding-right: 0px;
}

<body>
  <div>
    <div
      data-overlay-container="true"
    >
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            <label
              class="emotion-3"
              data-testid="favoriteCheese_label"
              id="react-aria-5"
            >
              Favorite cheese
            </label>
            <div
              aria-labelledby="react-aria-5"
              aria-orientation="vertical"
              id="react-aria-4"
              role="radiogroup"
            >
              <label
                class="emotion-4"
              >
                <input
                  aria-labelledby="radio-group-2-a-label"
                  checked=""
                  class="emotion-5"
                  data-testid="favoriteCheese_a"
                  name="react-aria-6"
                  tabindex="0"
                  type="radio"
                  value="a"
                />
                <div>
                  <div
                    class="emotion-6"
                    id="radio-group-2-a-label"
                  >
                    Asiago
                  </div>
                </div>
              </label>
              <label
                class="emotion-4"
              >
                <input
                  aria-labelledby="radio-group-2-b-label"
                  class="emotion-8"
                  data-testid="favoriteCheese_b"
                  name="react-aria-6"
                  tabindex="0"
                  type="radio"
                  value="b"
                />
                <div>
                  <div
                    aria-describedby="radio-group-2-b-description"
                    class="emotion-6"
                    id="radio-group-2-b-label"
                  >
                    Burratta
                  </div>
                  <div
                    class="emotion-10"
                    id="radio-group-2-b-description"
                  >
                    Burrata is an Italian cow milk cheese made from mozzarella and cream.
                  </div>
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          class="emotion-11"
        >
          <div
            style="height: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
